cli/config/rules.go

```diff
@@ -0,0 +1,187 @@
+	for _, cluster := range sortedMapValues(d.Clusters) {
+		for _, group := range sortedMapValues(cluster.Groups) {
+			if previousClusterName, ok := groupNameMap[group.Name]; ok {
+				return NewClusterConfigError(cluster.Name,
+					"the PXF service group '%s' is defined for PXF clusters '%s' and '%s', "+
+					"but must belong to only one PXF cluster", group.Name, previousClusterName, cluster.Name)
+			}
+			groupNameMap[group.Name] = cluster.Name
+		}
+	}
+	return nil
+}
+
+// There MUST be no more than one collocated PXF cluster
+func pxfDeploymentNoMoreThanOneCollocated(d *PxfDeployment) error {
+	previousClusterName := ""
+	for _, cluster := range sortedMapValues(d.Clusters) {
+		if cluster.Collocated {
+			if previousClusterName != "" {
+				return NewClusterPropertyConfigError(cluster.Name, "cluster.collocated",
+					"PXF clusters '%s' and '%s' are both defined as collocated, "+
+					"but there must be no more than one collocated PXF cluster", previousClusterName, cluster.Name)
+			}
+			previousClusterName = cluster.Name
+		}
+	}
+	return nil
+}
+
+// A collocated PXF cluster SHOULD NOT have a list of hosts, and they will be ignored, and a warning will be generated
+func pxfClusterIgnoreHostsWhenCollocatedWarning(c *PxfCluster) error {
+	if c.Collocated && len(c.Hosts) != 0 {
+		gplog.Warn("the PXF cluster '%s' is collocated, the provided host list will be ignored", c.Name)
+	}
+	return nil
+}
+
+// A collocated PXF cluster SHOULD NOT have an endpoint, and it will be ignored, and a warning will be generated
+func pxfClusterIgnoreEndpointWhenCollocatedWarning(c *PxfCluster) error {
+	if c.Collocated && c.Endpoint != "" {
+		gplog.Warn("the PXF cluster '%s' is collocated, the provided endpoint '%s' will be ignored", c.Name, c.Endpoint)
+	}
+	return nil
+}
+
+// An external cluster MUST have either a non-empty list of hosts or an endpoint or both
+func pxfClusterEndpointOrHostsMustExistWhenExternal(c *PxfCluster) error {
+	if !c.Collocated && c.Endpoint == "" && len(c.Hosts) == 0 {
+		return NewClusterConfigError(c.Name,
+			"the PXF cluster '%s' is not collocated, "+
+			"either a 'cluster.endpoint' property or a list of hosts (in hosts.txt file) must be defined", c.Name)
+	}
+	return nil
+}
+
+// A non-empty endpoint of a PXF cluster MUST have a value of either a valid IPv4 address or FQDN
+func pxfClusterValidEndpoint(c *PxfCluster) error {
+	if c.Endpoint != "" && !IsValidAddress(c.Endpoint) {
+		return NewClusterPropertyConfigError(c.Name, "cluster.endpoint",
+			"the endpoint '%s' of the PXF cluster '%s' is not a valid IPv4 address "+
+			"or FQDN", c.Endpoint, c.Name)
+	}
+	return nil
+}
+
+// A PXF cluster MUST have at least one PXF service group
+func pxfClusterAtLeastOneServiceGroup(c *PxfCluster) error {
+	if len(c.Groups) == 0 {
+		return NewClusterPropertyConfigError(c.Name, "group.[group_name].name",
+			"the PXF cluster '%s' has no PXF service group definitions, "+
+			"there must be at least one", c.Name)
```

Comment:
```suggestion
			"there must be at least one service group defined", c.Name)
```
For consistency with the function below